src/Suave/Http.fs

```diff
@@ -786,3 +786,81 @@ module Http =
             }
       }
       |> succeed
+
+  module CORS =
+    
+    open System
+    open Successful
+
+    [<Literal>]
+    let origin = "Origin"
+    [<Literal>]
+    let accessControlRequestMethod = "Access-Control-Request-Method"
+    [<Literal>]
+    let accessControlRequestHeaders = "Access-Control-Request-Headers"
+    [<Literal>]
+    let accessControlAllowOrigin = "Access-Control-Allow-Origin"
+    [<Literal>]
+    let accessControlAllowMethods = "Access-Control-Allow-Methods"
+    [<Literal>]
+    let accessControlAllowHeaders = "Access-Control-Allow-Headers"
+    [<Literal>]
+    let accessControlAllowCredentials = "Access-Control-Allow-Credentials"
+    [<Literal>]
+    let accessControlExposeHeaders = "Access-Control-Expose-Headers"
+    [<Literal>]
+    let accessControlMaxAge = "Access-Control-Max-Age"
+
+    let cors (allowedUris : string list) : WebPart =
+        fun (ctx : HttpContext) ->
+            let req = ctx.request
+            match req.header origin with
+            | Choice1Of2 originValue -> // CORS request
+                let allowedOrigin = List.exists (fun a -> a = originValue) allowedUris
+                match req.``method`` with
+                | HttpMethod.OPTIONS ->
+                    match req.header accessControlRequestMethod with
+                    | Choice1Of2 requestMethodHeaderValue -> // Preflight request
+                        
+                        // Is request method valid? Get from configuration? If no, don't set any headers and continue
+
+                        // Does the request have an Access-Control-Request-Headers header? If so, validate. If not, proceed.
+                        let setAccessControlRequestHeaders =
+                            match req.getAllHeaders accessControlRequestHeaders with
+                                | Choice1Of2 list -> 
+                                    Writers.setHeader accessControlAllowHeaders (list |> String.concat ", ")
+                                | Choice2Of2 _ -> succeed
+
+                        if allowedOrigin then
+                            ctx |>
+                                (
```

Comment:
It would be easier to read this if you assign the constructed web part to a constant and then use that with ctx.